packages/block-editor/src/store/private-selectors.js

```diff
@@ -547,3 +547,17 @@ export const getBlockStyles = createSelector(
 export function isZoomOutMode( state ) {
 	return state.editorMode === 'zoom-out';
 }
+
+/**
+ * Returns the clientID of the element that is
+ * considered to be the root for inserting new
+ * "sections" into the editor. This can be thought
+ * of as the "content" of the current editor.
+ *
+ * @param {Object} state Editor state.
+ *
+ * @return {string} The client ID of the section root.
+ */
+export function getSectionRootClientId( state = '' ) {
+	return state.sectionRootClientId;
+}
```

Comment:
```suggestion
export function getSectionRootClientId( state = '' ) {
let sectionRootClientId = state.sectionRootClientId;

	// Default to the main block if no section root is set.
	if ( ! sectionRootClientId ) {
		const blocks = getBlocksByName( state, 'core/group' );
		sectionRootClientId =
			blocks.find(
				( clientId ) =>
					getBlockAttributes( state, clientId )?.tagName === 'main'
			) ?? '';
	}

	return sectionRootClientId;
}
```

I wonder if we can default the the algorithm which prefers the `<main>` tag?

Then if an individual editor needs to customise the behaviour it can dispatch the `setSectionRootClientId()` action.